keras/applications/regnet.py

```diff
@@ -0,0 +1,1349 @@
+  https://keras.io/api/applications/#usage-examples-for-image-classification-models).
+
+  For transfer learning use cases, make sure to read the
+  [guide to transfer learning & fine-tuning](
+    https://keras.io/guides/transfer_learning/).
+
+  Note: Each Keras Application expects a specific kind of input preprocessing.
+  For Regnets, preprocessing is included in the model using a `Rescaling` layer.
+  RegNet models expect their inputs to be float or uint8 tensors of pixels with 
+  values in the [0-255] range.
+
+  The naming of models is as follows: `RegNet{block_type}{flops}` where 
+  `block_type` is one of `(Y, Z)` and `flops` signifies hundred million 
+  floating point operations. For example RegNetY64 corresponds to RegNet with 
+  Y block and 6.4 giga flops (64 hundred million flops). 
+
+  Args:
+    include_top: Whether to include the fully-connected
+        layer at the top of the network. Defaults to True.
+    weights: One of `None` (random initialization),
+          "imagenet" (pre-training on ImageNet),
+          or the path to the weights file to be loaded. Defaults to "imagenet".
+    input_tensor: Optional Keras tensor
+        (i.e. output of `layers.Input()`)
+        to use as image input for the model.
+    input_shape: Optional shape tuple, only to be specified
+        if `include_top` is False.
+        It should have exactly 3 inputs channels.
+    pooling: Optional pooling mode for feature extraction
+        when `include_top` is `False`. Defaults to None.
+        - `None` means that the output of the model will be
+            the 4D tensor output of the
+            last convolutional layer.
+        - `avg` means that global average pooling
+            will be applied to the output of the
+            last convolutional layer, and thus
+            the output of the model will be a 2D tensor.
+        - `max` means that global max pooling will
+            be applied.
+    classes: Optional number of classes to classify images
+        into, only to be specified if `include_top` is True, and
+        if no `weights` argument is specified. Defaults to 1000 (number of
+        ImageNet classes).
+    classifier_activation: A `str` or callable. The activation function to use
+        on the "top" layer. Ignored unless `include_top=True`. Set
+        `classifier_activation=None` to return the logits of the "top" layer.
+        Defaults to "softmax".
+        When loading pretrained weights, `classifier_activation` can only
+        be `None` or `"softmax"`.
+
+  Returns:
+    A `keras.Model` instance.
+"""
+
+def PreStem():
+  """Rescales and normalizes inputs to [0,1] and ImageNet mean and std.
+  
+  Args:
+    x: input tensor
+    
+  Returns:
+    Rescaled and normalized tensor
+  """
+
+  def apply(x):
+    IMGNET_MEAN = tf.constant([0.485, 0.456, 0.406])
+
+    IMGNET_STD = tf.constant([0.229, 0.224, 0.225])
+
+    x = layers.Rescaling(scale=1./255., name="prestem_rescaling")(x)
```

Comment:
Hardcoding names is problematic, because the block (`PreStem()`) is indented to be reused multiple times. Normally you'd want layer names to be unique, and if they're hardcoded here they will be reused every time the block is called. I'd recommend removing names entirely. If names are necessary then I'd recommend using a naming prefix system and a `name` argument in `PreStem()`.

This comment applies to other blocks as well.

Reply:
I've used a system for rest of the names. I'll add the "name" parameter here also, just like rest of the blocks.